### PR TITLE
Web: add batching support

### PR DESCRIPTION
**Pull request by kjcho-msft**

### CODE REVIEW

File: Website/Packages/datax-pipeline/package.json [hunk 1]
```diff
@@ -21,7 +21,9 @@
     "dev": "rimraf dist && webpack --mode development --config webpack.config.js --bail --progress --display-modules --display-error-details --colors --display-chunks --display-reasons",
     "devpatch": "rimraf dist && webpack --mode development --config webpack.config.js --bail --progress --display-modules --display-error-details --colors --display-chunks --display-reasons && localdevpatch.bat"
   },
-  "dependencies": {},
+  "dependencies": {
+    "react-datetime-picker": "^2.4.0"
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/package.json]: Remove the ^ from the version. We want to lock down to this specific version that you have tested as opposed to opening up the versions to picking up latest version which may have bugs or incompabilities.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/api.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ import * as Helpers from './flowHelpers';
 // Flow Service
 export const getFlow = name => getProduct(name).then(f => f.gui);
 
-export const saveFlow = config =>
+export const deployFlow = config =>
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/api.js]: Is this a bug by renaming method to deployFlow? Where is saveFlow going? I ask because below, we call the flow/save service API. Why is it not calling something like flow/deploy?

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx [hunk 3]
```diff
@@ -318,7 +351,42 @@ class FlowDefinitionPanel extends React.Component {
         );
     }
 
+    renderScheduleTab() {
+        if (this.props.input.mode === Models.inputModeEnum.streaming){
+            return null;
+        }
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: can you make sure you open the datax-pipeline folder in VS Code and save any open files to get Prettier to kick in to help you style our codebase consistently. I can tell by looking at the code where this should have looked like "} else {" automatically. I assume you opened the Website\Packages folder which is why you didn't get auto formatting on save operations. If you want to open the parent folder like this, try adding a .prettierrc file at that folder level to solve this going forward.

Comment (barryt2) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: i.e. add .prettierrc file to Website\Packages folder. I haven't tried it so not sure if it will work but give it a try. You should look online for a solution, I'm sure its supported in some way.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx [hunk 4]
```diff
@@ -318,7 +351,42 @@ class FlowDefinitionPanel extends React.Component {
...
+            return (                               
+                <VerticalTabItem linkText="Schedule" itemCompleted={this.props.scheduleValidated}>
+                <ScheduleSettingsContent
+                    batchList={this.props.flow.batchList}
+                    selectedBatchIndex={this.props.selectedBatchIndex}
+                    onNewBatch={this.props.onNewBatch}
+                    onDeleteBatch={this.props.onDeleteBatch}
+                    onUpdateSelectedBatchIndex={this.props.onUpdateSelectedBatchIndex}
+                    addBatchButtonEnabled={this.state.addBatchButtonEnabled}
+                    deleteBatchButtonEnabled={this.state.deleteBatchButtonEnabled}
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: nit: out of curiosity, I see in most of the files I've reviewed so far for the new batch support that you add newlines above and below your newly added members. Can you remove these new blank lines? Its not needed and if future folks were to keep doing this for every feature they add, it will just add a bunch of unnecessary blank lines sprinkled across our codebase.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx [hunk 5]
```diff
@@ -593,11 +676,41 @@ class FlowDefinitionPanel extends React.Component {
...
         });
 
         this.props
             .onSaveFlow(this.props.flow)
+            .then(name => {
+                this.setState({
+                    isSaving: false,
+                    showMessageBar: true,
+                    messageBarIsError: false,
+                    messageBarStatement: 'Flow definition is saved. If there are any scheduled batch jobs in thie flow, they will be created and started by the scheduler.'
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: I recommend you provide different message bar statements based on first checking if the flow input type is batch versus streaming. Its best to provide messages that are contextual. i.e. I find it odd that we mention about batch even for the streaming case. Lets not make our customers think since we can provide more contextual messages. BTW, this above is spelled incorrectly (i.e. thie flow)

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx [hunk 6]
```diff
@@ -805,15 +921,23 @@ const mapDispatchToProps = dispatch => ({
     onUpdateInputMode: mode => dispatch(Actions.updateInputMode(mode)),
     onUpdateInputType: type => dispatch(Actions.updateInputType(type)),
     onUpdateInputHubName: name => dispatch(Actions.updateInputHubName(name)),
+    onUpdateInputPath: path => dispatch(Actions.updateInputPath(path)),
     onUpdateInputHubConnection: connection => dispatch(Actions.updateInputHubConnection(connection)),
     onUpdateInputSubscriptionId: id => dispatch(Actions.updateInputSubscriptionId(id)),
     onUpdateInputResourceGroup: name => dispatch(Actions.updateInputResourceGroup(name)),
+    
+    onupdateBatchInputConnection: connection => dispatch(Actions.updateBatchInputConnection(connection)),
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: typo: capitalize the U in onUpdateBatchInputConnection to be consistent with other methods you have defined below and in this file.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/rule/rulesSettingsContent.jsx [hunk 7]
```diff
@@ -319,8 +319,7 @@ const rightPaneStyle = {
 const listContainerStyle = {
     backgroundColor: Colors.white,
     border: `1px solid ${Colors.neutralTertiaryAlt}`,
-    flex: 1,
-    overflowY: 'auto'
+    flex: 1
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/rule/rulesSettingsContent.jsx]: is this change intentional? this UI is related to rules and not batching/scheduling, did you reuse this PR to fix some other bug? By remove the overflowY: 'auto' statement, what does this fix?

File: Website/Packages/datax-pipeline/package.json (hunk 1 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/package.json]: updated.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/api.js (hunk 2 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/api.js]: renamed to saveFlow

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+                <DateTimePicker
+                    className="ms-font-m info-settings-textbox"
+                    value={this.props.batch.properties.endTime}
+                    onChange={(value) => this.props.onUpdateBatchEndTime(value)}
+                />
+            </div>
+        );
+    }
+
+    renderInterval() {
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx]: Looking at your shared screenshots of the UI, I don't like how you are rendering the width of the sections for Recurrence, Delay, Window (in Recurring Job Settings) and Interval, Window (in One Time Job Settings). If user has wide monitors and most do, you are rendering the text box and drop downs to take entire width of the settings panel. Lets not do this and fix the width. For example, in all cases for the text box, you simply accept a number and in all cases, the Unit drop down is just a short text, I I think you can make the width of these section container style reuse the same width specified by Alias, Batch Type, Start Time, End Time, etc. This way both controls will take up 50% of width where the width is consitent with the other controls in the UI.

Comment (barryt2) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx]: please make sure you do this for both the one time and recurring job settings.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+        return (
+            <div style={sectionContainerStyle}>
+                {this.renderWindowText()}
+                {this.renderWindowDropdown()}
+            </div>
+        );
+    }
+
+    renderWindowText() {
+        return (
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx]: prettier would have formatted this correctly on save. you just need to open the datax-pipeline package folder to kick in. See my other comments I added previously.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+                    spellCheck={false}
+                    label="Interval"
+                    value={this.props.batch.properties.interval}
+                    onChange={(event, value) => this.props.onUpdateBatchIntervalValue(value)}
+                />
+            </div>
+        );
+    }
+
+    renderIntervalTypeDropdown() {
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx]: The entire definition of renderIntervalTypeDropdown is repeated/duplicated 5 times in your latest changes, why not just introduce one function called renderUnitTypeDropdown(type, onUpdateType)? i.e. you can call this method passing in different 'type' value parameter and 'onUpdateType' function parameter. There is no need to duplicate the code 5 times.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+                    disabled={true}
+                    value={this.props.batchTypeDisplayName}
+                />
+            </div>
+        );
+    }
+
+    renderTimeRange() {
+        return (
+            <div >
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx]: prettier :-)

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+                    className="ms-font-m"
+                    options={options}
+                    selectedKey={this.props.batch.properties.intervalType}
+                    onChange={(event, selection) => this.props.onUpdateBatchIntervalType(selection.key)}
+                />
+            </div>
+        );
+    }
+
+    renderDelay() {
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx]: please see my remarks in one time job settings. We don't need 3 functions duplicated here

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx [hunk 13]
```diff
@@ -0,0 +1,329 @@
...
+    },
+    {
+        key: 'columnBatchStatus',
+        name: 'Status',
+        fieldName: 'status',
+        isResizable: true
+    }
+];
+
+export default class scheduleSettingsContent extends React.Component {
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx]: you should always capitolize the first letter of React components.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx [hunk 14]
```diff
@@ -0,0 +1,329 @@
...
+
+const batchColumns = [
+    {
+        key: 'columnBatchName',
+        name: 'Alias',
+        fieldName: 'id',
+        isResizable: true
+    },
+    {
+        key: 'columnBatchType',
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx]: please look at the office-ui-fabric documentations for the DetailsList control, I recommend the name column take 50% of the space and Type and Status columns each take 25%

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx [hunk 15]
```diff
@@ -0,0 +1,329 @@
...
+    }
+
+    renderButtons() {
+        return [this.renderAddBatchButton(), this.renderDeleteBatchButton()];
+    }
+
+    renderAddBatchButton() {
+        let recurringExists = this.props.batchList.filter(b => b.type  === Models.batchTypeEnum.recurring);
+
+        const menuItems = (recurringExists === undefined || recurringExists.length < 1)
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx]: FYI: recurringExists === undefined is equivalent to !recurringExists. I recommend to use the latter.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx (hunk 4 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: Prettier fixed this :)

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx (hunk 3 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: Right. I have to open Packages\datax-pipleline folder.  I tried to add .prettierrc file to Website\Packages folder but it didn't work.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js [hunk 16]
```diff
@@ -447,21 +447,25 @@ export function convertFlowToConfig(flow) {
...
     let rules = [...flow.rules];
+    let batchList = [...flow.batchList];
 
     // sort by name
     referenceData.sort((a, b) => a.id.localeCompare(b.id));
     functions.sort((a, b) => a.id.localeCompare(b.id));
     sinkers.sort((a, b) => a.id.localeCompare(b.id));
     outputTemplates.sort((a, b) => a.id.localeCompare(b.id));
     rules.sort((a, b) => a.id.localeCompare(b.id));
+
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: remove empty new line :-)

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js [hunk 17]
```diff
@@ -471,10 +475,17 @@ export function convertFlowToConfig(flow) {
...
         outputTemplates: outputTemplates,
-        rules: convertFlowToConfigRules(rules)
+        rules: convertFlowToConfigRules(rules),
+        batchList: batchList,
+        version: getTimeStamp()
     };
 }
 
+function getTimeStamp() {
+    let timeStamp = (new Date()).toISOString().slice(0, 19).replace(/-/g, "").replace(/:/g, "").replace("T", "");
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: prettier would have converted your double quotes to single quotes for you. Please reopen all 16 files and click save for each (or Save All) to have all your files converted to consistent format configured for this package.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx (hunk 6 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: fixed.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js (hunk 17 above)
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: what are you doing here for the timeStamp? Can you add a comment as its not clear what all the magic replace is doing. BTW, why are you not using the "Moment" widely used 3rd datetime library? It would help address a lot of the common date operations/formating/assignments/conversions.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js [hunk 18]
```diff
@@ -3,17 +3,40 @@
...
+};
+
+export const jobModeEnum = {
+    recurrence: 'recurrence',
+    nonrecurrence: 'nonrecurrence',
+};
+
+export const inputFormatTypeEnum = {
+    json: 'json',
+    parquet: 'parquet'
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js]: did you actually add support for parquet or is this work still pending? If not supported or soon to be supported shortly afterwards, I would recommend you to remove it until the support is about to exist.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/rule/rulesSettingsContent.jsx (hunk 7 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/rule/rulesSettingsContent.jsx]: no, it wasn't intentional. reverted.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx (hunk 9 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx]: done.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js (hunk 18 above)
Comment (barryt2) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js]: nevermind, I see you don't expose it to the UI dropdown, this is just a enum. Ignore what I said. You did the right thing here.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx (hunk 11 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx]: done

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js [hunk 19]
```diff
@@ -527,6 +645,44 @@ export function getMetricSinker() {
...
+        return {
+            id: '',
+            type: type,
+            disabled: false,
+            properties: {
+                interval: '1',
+                intervalType: 'day',
+                delay: '0',
+                delayType: 'day',
+                window: '1',
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js]: out of curiosity, why do we store all the numbers as strings? is this a backend limitations? otherwise, I would have expected the number fields being just a number as opposed to a string.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx (hunk 13 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx]: fixed.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx (hunk 15 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx]: done

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js (hunk 16 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: done

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js [hunk 20]
```diff
@@ -26,9 +27,14 @@ const INITIAL_FLOW_STATE = {
     outputTemplates: [],
     rules: [],
 
+    batchInputs: [Models.getDefaultBatchInputSettings()],
+    batchList: [],
+
     // State
     isNew: false,
     isDirty: false,
+    selectedFlowBatchInputIndex: undefined,
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js]: For selectedFlowBatchInputIndex, besides seeing it set in FLOW_INIT, its actually not used anywhere else. I think you can remove this unneeded field as selectBatchIndex is the one you are actually using.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js [hunk 21]
```diff
@@ -49,6 +55,7 @@ export default (state = INITIAL_FLOW_STATE, action) => {
             return Object.assign({}, INITIAL_FLOW_STATE, flow, {
                 isNew: false,
                 isDirty: false,
+                selectedFlowBatchInputIndex: flow.input.batch && flow.input.batch.length > 0 ? 0 : undefined,
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js]: bug: use selectedBatchIndex here instead

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js [hunk 22]
```diff
@@ -63,6 +70,7 @@ export default (state = INITIAL_FLOW_STATE, action) => {
                 isDirty: true,
                 owner: action.payload,
                 selectedSinkerIndex: 0, // new flow by default contains the metric sinker
+                selectedFlowBatchInputIndex: 0,
```
Comment (barryt2) [inline Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js]: bug: use selectedBatchIndex here instead.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js (hunk 17 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: no I didn't add this. it was probably added by accident when I merged with master. I removed this.

Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowHelpers.js]: no I didn't add this. it was probably added by accident when I merged with master. I removed this.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js (hunk 18 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js]: cool

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js (hunk 20 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js]: I use selectedFlowBatchInputIndex.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js (hunk 19 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js]: no special reason. but I think we always handle the numbers this way. e.g. windowDuration and maxRate

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js (hunk 21 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js]: selectedFlowBatchInputIndex is correct. this is an index to support multiple inputs. and selectedBatchIndex is an index for the batch job list.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js (hunk 22 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/flowReducer.js]: selectedFlowBatchInputIndex is correct. this is an index to support multiple inputs. and selectedBatchIndex is an index for the batch job list.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx (hunk 5 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx]: fixed. show this message only if it is a batch mode.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx (hunk 8 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx]: done. use the same width.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx (hunk 10 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/oneTimeScheduleSettings.jsx]: done. created the help methods and used them.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx (hunk 12 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx]: Done. created/used the helper methods.

File: Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx (hunk 14 above)
Comment (kjcho-msft) [reply Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/scheduleSettingsContent.jsx]: I made this look better by setting the miWidth for typeDisplay and Status columns.
